safe_transaction_service/history/indexers/tx_processor.py

```diff
@@ -163,13 +164,14 @@ def get_safe_version_from_master_copy(
 
     def get_last_safe_status_for_address(
         self, address: ChecksumAddress
-    ) -> Optional[SafeStatus]:
-        safe_status = self.safe_status_cache.get(
-            address
-        ) or SafeStatus.objects.last_for_address(address)
-        if not safe_status:
+    ) -> Optional[SafeLastStatus]:
+        try:
+            safe_status = self.safe_last_status_cache.get(
+                address
+            ) or SafeLastStatus.objects.get(address=address)
+            return safe_status
```

Comment:
Yes, because you can be processing a lot of traces, and you would have to query database for each one (this is already a optimization I did when profiling this method long time ago)